crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -103,6 +105,86 @@ use bevy_platform::sync::atomic::AtomicIsize as AtomicIdCursor;
 #[cfg(not(target_has_atomic = "64"))]
 type IdCursor = isize;
 
+/// This represents the row or "index" of an [`Entity`] within the [`Entities`] table.
+/// This is a lighter weight version of [`Entity`].
+///
+/// This is a unique identifier for an entity in the world.
+/// This differs from [`Entity`] in that [`Entity`] is unique for all entities total (unless the [`Entity::generation`] wraps),
+/// but this is only unique for entities that are active.
+///
+/// This can be used over [`Entity`] to improve performance in some cases,
+/// but improper use can cause this to identify a different entity than intended.
+/// Use with caution.
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Debug, Display)]
+#[cfg_attr(feature = "bevy_reflect", derive(Reflect))]
+#[cfg_attr(feature = "bevy_reflect", reflect(opaque))]
+#[cfg_attr(feature = "bevy_reflect", reflect(Hash, PartialEq, Debug, Clone))]
```

Comment:
```suggestion
#[cfg_attr(
    feature = "bevy_reflect",
    derive(Reflect),
    reflect(opaque),
    reflect(Hash, PartialEq, Debug, Clone)
)]
```

Nit: I would rather combine these just to reduce the quantity of conditional compilation. I consider each `feature = "..."` to be its own code smell that should be minimized as-much-as-possible.

Reply:
This is the same format as `Entity` and probably elsewhere. I personally don't have a preference here, as long as it's consistent. If there's a consensus to do it this way, I can just do it everywhere in this pr, otherwise, I think this would be better as a follow up PR so it can be discussed separately. Depends on how controversial this is.

Reply:
Oh yeah then just keep it as-is that's fine. I didn't see that it was already this way in some other spots.